### PR TITLE
Stateless collapse header

### PR DESCRIPTION
**Pull request by cybadave**
The use cases for CollapseHeader require that state be managed externally. 

The callback function expandCollapse() in the properties will suffice.

Property open has been renamed to expanded as [react implements open as a DOM attribute](https://reactjs.org/docs/dom-elements.html)

### CODE REVIEW

File: components/sections/CollapseHeader.js [hunk 1]
```diff
@@ -1,30 +1,16 @@
...
-        this.state = {
-            open: this.props.open
-        }
-    }
-
-    expandCollapse() {
-        this.setState({
-            open: !this.state.open
-        })
+        compact: false,
```
Comment (victorhtc) [inline components/sections/CollapseHeader.js]: We should implement a default `expandCollapse` behaviour here. I wouldn't remove it. If an `expandCollapse` prop is set, then the control would be external.

File: components/sections/CollapseHeader.js [hunk 2]
```diff
@@ -37,27 +23,22 @@ export default class CollapseHeader extends React.PureComponent {
...
             compact
         })
 
         return (
             <div className={headerClassName} style={style}>
-                <div
-                    className="m-collapse-label-container"
-                    onMouseOver={onMouseOver}
-                    onMouseLeave={onMouseLeave}
-                >
```
Comment (victorhtc) [inline components/sections/CollapseHeader.js]: Why removing these listeners?

**Comment by jimmed** [comment]
>  react implements open as a DOM attribute

TIL! This explains some irksome behaviour in the past...

File: components/sections/CollapseHeader.js (hunk 1 above)
Comment (jimmed) [reply components/sections/CollapseHeader.js]: I'm keen to offer both options - stateful and stateless, but I don't like the idea of controlling this via a prop.

I can't think of a scenario where we would want to control whether a given `CollapseHeader` was stateful or not, so there's no need to use a prop.

As an alternative, I would suggest we offer two components:

File: components/sections/CollapseHeader.js (hunk 2 above)
Comment (jimmed) [reply components/sections/CollapseHeader.js]: Likewise, I think we should keep these listeners - although we'd probably want to add more than just mouse over/leave if we do.

File: components/sections/CollapseHeader.js [hunk 3]
```diff
@@ -1,63 +1,45 @@
-import React from "react"
-import IconArrow from "../icons/IconArrow"
-import cx from "classnames"
-import noop from "lodash/noop"
+import React from 'react';
+import IconArrow from '../icons/IconArrow';
+import cx from 'classnames';
+import noop from 'lodash/noop';
 
 export default class CollapseHeader extends React.PureComponent {
```
Comment (diogobeda) [inline components/sections/CollapseHeader.js]: now that it's stateless and has only a render function, this could be rewritten as a function component

Comment (cybadave) [reply components/sections/CollapseHeader.js]: :+1:

**Pull request merged**